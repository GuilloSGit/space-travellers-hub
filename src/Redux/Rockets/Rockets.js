import { Action } from "history";

export const FETCH_ROCKETS = 'space-travelers-hub/Rockets/FETCH_ROCKETS';
export const url = 'https://api.spacexdata.com/v3/rockets';
export const initialState = [];

//on fetched book 

const onSuccess = (rockets) => ({
    type: FETCH_ROCKETS,
    playload: rockets
})

//Fetched rocket action

const fetchRocketsAction = () => async (dispatch) =>{
    const response = await fetch(url);
    const rocketsData = await response.json();
    const fetchedRockets = rocketsData.map((rocket) => ({
        id: rocket[0],
        rocket_name: rocket[0].rocket_name,
        description: rocket[0].flickr_images
    }))
    dispatch(onSuccess(fetchedRockets));
}

const rocketsReducer = (state = initialState, action) => {
    
}

export default fetchRocketsAction;