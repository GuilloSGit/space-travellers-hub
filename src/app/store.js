import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from '../redux/missions/missions';
import rocketReducer from '../redux/rockets/Rockets';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketReducer,
  },
});

export default store;
