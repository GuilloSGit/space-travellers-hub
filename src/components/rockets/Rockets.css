.cards-container{
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: repeat(3, 1fr);
    gap: 30px;
}

h1{
    color: #000;
    font-weight: bold;
}

.card{
    display: grid;
    grid-template-columns: 280px 1fr;
}

.rocket-img, .rocket-description{
    margin: 10px;
}

.img{
    width: 100%;
    height: 100%;
}

.reserve-rocket{
    background-color: #007bff;
    padding: 9px;
    font-size: 14px;
    border-radius: 3px;
    color: rgba(255, 255, 255, 0.86);
}

.reserve-rocket:hover{
    cursor: pointer;
    color: #fff;
}

.description{
    color: rgb(63, 60, 60);
}

.canceled{
    border : none;
    border-radius: 4px;
    padding: 3px;
    height: max-content;
    background-color: #007bff;
    color: #fff;
}

.cancel-reservation{
    border: 2px solid #8c8e90;
    color: #8c8e90;
    border-radius: 3px;
}

.cancel-reservation:hover{
    cursor: pointer;
}