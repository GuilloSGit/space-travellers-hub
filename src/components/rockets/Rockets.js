import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchedRockets, updateRockets } from '../../redux/rockets/Rockets';
import './Rockets.css';

const Rockets = () => {
  const dispatch = useDispatch();
  const allRockets = useSelector((state) => state.rockets);

  useEffect(() => {
    if (!allRockets.length) dispatch(fetchedRockets());
  }, [dispatch]);

  const handleClick = ({ target }) => {
    const { id } = target;
    dispatch(updateRockets(Number(id)));
  };

  return (
    <div className="cards-container">
      {allRockets.map((rocket) => ((
        <div key={rocket.id} className="card">
          <div className="rocket-img">
            <img className="img" src={rocket.flickr_image} alt="rocket" />
          </div>
          <div className="rocket-description">
            <h1>{rocket.rocket_name}</h1>
            <span />
            <p className="description">
              <span>{rocket.canceled && <button type="button" className="canceled">Reserved</button>}</span>
              {rocket.description}
            </p>
            {rocket.canceled && (
            <button
              id={rocket.id}
              onClick={handleClick}
              className="cancel-reservation"
              type="button"
            >
              Cancel Reservation
            </button>
            )}
            {!rocket.canceled && (
            <button
              id={rocket.id}
              onClick={handleClick}
              className="reserve-rocket"
              type="button"
            >
              Reserve Rocket
            </button>
            )}
          </div>
        </div>
      )))}
    </div>
  );
};

export default Rockets;
