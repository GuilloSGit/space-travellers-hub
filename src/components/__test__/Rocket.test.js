import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../app/store';
import Rockets from '../rockets/Rockets';

describe('Rockets Component test', () => {
  it('Renders Mission Row', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
