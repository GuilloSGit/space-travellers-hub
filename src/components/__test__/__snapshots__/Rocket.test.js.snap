// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets Component test Renders Mission Row 1`] = `
<div
  className="cards-container"
/>
`;
